shared/runtime/softtimer.h

```diff
@@ -73,12 +73,14 @@ static inline void soft_timer_reinsert(soft_timer_entry_t *entry, uint32_t initi
 #if !defined(MICROPY_SOFT_TIMER_TICKS_MS)
 // IF MICROPY_SOFT_TIMER_TICKS_MS is not defined then the port must provide the
 // following timer functions:
+// - soft_timer_init() to initialize soft timer hardware as needed.
```

Comment:
This is not really part of the global API, it's only specific to a port and doesn't need to be provided if it's not needed.

Reply:
Currently the only part that uses this approach is rp2, so the distinction between global and rp2 is a bit moot. ;)

Is it enough to add a note in the comment here that the function's existence is port-specific? Or would you rather I named it something else and put it in another header (if so, where?)

Reply:
> Is it enough to add a note in the comment here that the function's existence is port-specific?

Yes that's fine.